src/bokeh/core/property/pd.py

```diff
@@ -55,6 +60,29 @@ class PandasDataFrame(Property["DataFrame"]):
     def validate(self, value: Any, detail: bool = True) -> None:
         super().validate(value, detail)
 
+        if nw.dependencies.is_into_dataframe(value):
+            return
+
+        msg = "" if not detail else f"expected object convertible to Narwhals DataFrame, got {value!r}"
+        raise ValueError(msg)
+
+class PandasDataFrame(Property["DataFrame"]):
+    """ Accept Pandas DataFrame values.
+
+    .. deprecated:: 3.7.0
+        Use ``EagerDataFrame()`` instead.
+
+    This property only exists to support type validation, e.g. for "accepts"
+    clauses. It is not serializable itself, and is not useful to add to
+    Bokeh models directly.
+
+    """
+    def __init__(self) -> None:
+        deprecated((3,7,0), "PandasDataFrame", "EagerDataFrame")
```

Comment:
`PandasDataFrame` is presumably more specific than `EagerDataFrame` so I could see keeping this in case we ever need to make that specific distinction (maybe add comments to clarify that narrow scope though)